event-bus/cassandra/src/main/java/org/apache/james/events/CassandraEventDeadLetters.java

```diff
@@ -89,4 +111,26 @@ public Flux<Group> groupsWithFailedEvents() {
     public Mono<Boolean> containEvents() {
         return cassandraEventDeadLettersDAO.containEvents();
     }
+
+    private Event deserialize(String json) {
+        List<Event> events = allEventSerializers.stream()
+            .map(eventSerializer -> Optional.ofNullable(deserialize(json, eventSerializer)))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .toList();
+
+        switch (events.size()) {
+            case 0: throw new RuntimeException("Could not deserialize event: " + json);
+            case 1: return events.getFirst();
+            default: throw new RuntimeException("More than one serializers could deserialize event: " + json);
+        }
```

Comment:
Is this a check we need to do at runtime or we could do it at startup?

Because then we could just findFirst...orElseThrow

Reply:
Really difficult to check at startup

Although there is a very small chance this case happen, I still want to check this case just for precaution.

Reply:
removed